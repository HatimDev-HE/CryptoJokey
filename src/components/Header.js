import React from 'react';
import { Routes, Route } from 'react-router-dom';
import Details from './Details';
import Home from './Home';
// import mic from './Assets/mic.svg';
// import gear from './Assets/gear.svg';
import Nav from './Nav';

function Header() {
  return (
    <>
      <div>
        <Nav />
        <Routes>
          <Route path="/" element={<Home />} />
          <Route path="/details" element={<Details />} />
        </Routes>
      </div>

    </>
  );
}

export default Header;
