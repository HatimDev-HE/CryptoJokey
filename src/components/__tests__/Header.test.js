import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../../redux/configstore';
import Header from '../Header';

describe('testing Header comp', () => {
  test('Header to match snapshot', () => {
    const tree = renderer.create(
      <BrowserRouter>
        <Provider store={store}>
          <Header />
        </Provider>
      </BrowserRouter>,
    );
    expect(tree).toMatchSnapshot();
  });
});
