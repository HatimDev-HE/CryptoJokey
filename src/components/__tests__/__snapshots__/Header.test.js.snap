// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing  comp Nav to match snapshot 1`] = `
<div>
  <div
    className="topnav"
  >
    <li
      className="navHome"
    >
      <a
        aria-current="page"
        className="active"
        href="/"
        onClick={[Function]}
      >
        <img
          alt=""
          height={30}
          src="caret-left.svg"
          width={30}
        />
      </a>
    </li>
    <li
      className="topHome"
    >
      JokeR
    </li>
    <li>
      <img
        alt=""
        height={30}
        src="mic.svg"
        width={30}
      />
    </li>
    <li>
      <img
        alt=""
        height={30}
        src="gear.svg"
        width={30}
      />
    </li>
  </div>
  <div
    className="headall"
  >
    <header>
      Welcome to CryptoJokey
    </header>
    <p
      className="slogan"
    >
      ‘In crypto we roll‘
    </p>
  </div>
  <div>
    <div
      className="headall"
    >
      <input
        id="search"
        name="search"
        onChange={[Function]}
        placeholder="Search Jokey"
        type="search"
      />
      <p
        className="jokey-coins"
      >
        All Jokey Coins
      </p>
    </div>
  </div>
</div>
`;

exports[`testing Header comp Header to match snapshot 1`] = `
<div>
  <div
    className="topnav"
  >
    <li
      className="navHome"
    >
      <a
        aria-current="page"
        className="active"
        href="/"
        onClick={[Function]}
      >
        <img
          alt=""
          height={30}
          src="caret-left.svg"
          width={30}
        />
      </a>
    </li>
    <li
      className="topHome"
    >
      JokeR
    </li>
    <li>
      <img
        alt=""
        height={30}
        src="mic.svg"
        width={30}
      />
    </li>
    <li>
      <img
        alt=""
        height={30}
        src="gear.svg"
        width={30}
      />
    </li>
  </div>
  <div
    className="headall"
  >
    <header>
      Welcome to CryptoJokey
    </header>
    <p
      className="slogan"
    >
      ‘In crypto we roll‘
    </p>
  </div>
  <div>
    <div
      className="headall"
    >
      <input
        id="search"
        name="search"
        onChange={[Function]}
        placeholder="Search Jokey"
        type="search"
      />
      <p
        className="jokey-coins"
      >
        All Jokey Coins
      </p>
    </div>
  </div>
</div>
`;

exports[`testing Nav comp Nav to match snapshot 1`] = `
Array [
  <div
    className="topnav"
  >
    <li
      className="navHome"
    >
      <a
        aria-current="page"
        className="active"
        href="/"
        onClick={[Function]}
      >
        <img
          alt=""
          height={30}
          src="caret-left.svg"
          width={30}
        />
      </a>
    </li>
    <li
      className="topHome"
    >
      JokeR
    </li>
    <li>
      <img
        alt=""
        height={30}
        src="mic.svg"
        width={30}
      />
    </li>
    <li>
      <img
        alt=""
        height={30}
        src="gear.svg"
        width={30}
      />
    </li>
  </div>,
  <div
    className="headall"
  >
    <header>
      Welcome to CryptoJokey
    </header>
    <p
      className="slogan"
    >
      ‘In crypto we roll‘
    </p>
  </div>,
]
`;
